src/main/kotlin/week1/racingcar/RacingGame.kt

```diff
@@ -1,36 +1,62 @@
 package week1.racingcar
 
-class RacingGame {
+import kotlin.random.Random
 
-    private var numberOfCars = 0
-    private var timesToRace = 0
+class RacingGame {
 
-    fun start() {
-        InputView().startInput().also {
-            this@RacingGame.apply {
-                numberOfCars = it.numberOfCars
-                timesToRace = it.timesToRace
-            }
-            race()
+    fun getInputResult(): InputResult {
+        return InputView().let {
+            it.getValidateRaceInfo(it.startInput())
         }
     }
 
-    private fun race() {
+    fun startGame(inputResult: InputResult) {
+        val cars = inputResult.cars.map { Car(it) }
+        val timesToRace = inputResult.timesToRace
+        race(cars, timesToRace)
+    }
+
+    private fun race(cars: List<Car>, timesToRace: Int) {
         val resultView = ResultView()
-        val cars = List(numberOfCars) { Car() }
-        (0 until timesToRace).forEach { _ ->
-            cars.forEach {
-                if (isPossibleToDrive()) {
-                    it.go()
-                }
-            }
-            resultView.showResult(cars)
+        repeat(timesToRace) {
+            drive(cars)
+            showResult(resultView, cars)
+        }
+    }
+
+    private fun drive(cars: List<Car>) {
+        cars.forEach {
+            if (isPossibleToDrive()) it.go()
+        }
+    }
+
+    private fun showResult(resultView: ResultView, cars: List<Car>) {
+        resultView.run {
+            showResult(cars)
+            showWinner(getWinners(cars))
         }
     }
 
-    private fun isPossibleToDrive() = RandomUtils.getRandomDigit() > 4
+    private fun isPossibleToDrive() = Random.nextInt(COUNT_OF_DIGITS) >= MIN_NUM_TO_GO
+
+    private fun getWinners(cars: List<Car>): List<String> {
+        val farthestMileage = getFarthestMileage(cars)
+        return cars.filter { it.mileage == farthestMileage }.map { it.name }
```

Comment:
객체의 프로퍼티를 외부로 꺼내서 직접 비교하는 것이 아니라 아래와 같이 객체에게 메시지를 보내보면 어떨까요? 🤗
```suggestion
        return cars.filter { it.isSameMileage(farthestMileage) }.map { it.name }
```

아래의 글을 참고해보시면 좋을 것 같아요. 😃
[getter를 사용하는 대신 객체에 메시지를 보내자](https://tecoble.techcourse.co.kr/post/2020-04-28-ask-instead-of-getter/)